optking/molsys.py

```diff
@@ -636,15 +639,15 @@ def interfrag_dq_discontinuity_correction(self, dq):
         for iDI, DI in enumerate(self._dimer_intcos):
             DI.dq_discontinuity_correction(dq[self.dimerfrag_intco_slice(iDI)])
 
-    # Returns mass-weighted Bmatrix if masses are supplied.
-    def compute_b_mat(self, masses=None):
+    # Returns mass-weighted Bmatrix if use_masses is True.
+    def Bmat(self, massWeight=False):
```

Comment:
This would make a good pull request.  Bmat is a function of geometry (and possibly of masses if mass-weighted, which I don't think really matters for anything except possibly IRC).  But Bmat does need recomputed for any geometry change, including (usually and most rigorously) any cartesian change within the iterative back-transformations when taking steps dq->dx.

Reply:
Perhaps one could write an intermediary function that tests how many _identically_ valued B matrices get computed.  I'm really not sure.  If you transformed the Hessian, and then took a geometry step you would repeat a Bmatrix calculation, but that's fairly unusual.  Some people try to converge the back-transformation with a static B-matrix (which it is for an infinitesimal displacement but I've found better convergence power with recomputation of it.  Could be made a flag.  In general, I alway reckoned at some future date someone could try to run on something really big and work on bottlenecks.  I am unaware that computational power is an issue at this point for anything anyone would use optking for.